fluidsynth/src/sfloader/fluid_defsfont.h

```diff
@@ -435,7 +435,7 @@ struct _fluid_defpreset_t
 {
   fluid_defpreset_t* next;
   fluid_defsfont_t* sfont;                  /* the soundfont this preset belongs to */
-  char name[21];                        /* the name of the preset */
+  char name[22];                        /* the name of the preset */
```

Comment:
Hm, I agree with your first comment. Maybe we should really fix the actual `FLUID_SPRINTF`, where GCC assumes the overflow to appear. Of course the compiler assumes that the arguments `bank` and `prenum` can become up to 5 bytes in size, each. I'm currently not sure about the valid range of `bank` (max. 128 for drums?). But if I just read correctly in the spec, at least preset numbers are in the range 0..127.
So, can you confirm that the `prenum` value cannot get greater than 127? Then I would change that argument to `sfpreset->prenum % 128`. GCC will recognize this and know that the string will always fit. And I'll revert the array size change.